### DISCUSSION TITLE
Contributions?

### DISCUSSION THREAD

**Discussion by schambers** [fluentmigrator/fluentmigrator#1349] [General]
Hello,

I'm happy to see that this framework has evolved so much even though I haven't contributed for a very long time (~10 years?)

That being said, hacktoberfest is coming up and I'm curious if there's any easy-ish areas I can make contributions for some pull requests. I'm not sure I would recognize any of the code anymore or if it even has some of the original parts and I don't really do any .net coding anymore (java mainly now)

In addition, if there's any clerical work I can assist with (tagging issues for hacktoberfest or cleanup etc?) I'd love to help with that as well. 

Again, thank you for doing a great job running this. It makes me happy that so many people found it useful over the years. One of my current co-workers said he used it at his previous workplace and it worked very well. 

Thank you!!

Sean Chambers

**Comment by jzabroski** [comment]
Hi @schambers 

# Immediate Roadmap
I am in the process of creating FluentMigrator v5 (for .NET 5).

- #1218 Immediate help would be deleting all SqlAnywhere logic.  That's pretty easy but also a good way to get familiar with the code.
- I need to merge #1178 
- I need to merge all develop features into master so we can do GitFlow going forward #1348
- Need to check - I think #1259 can now be closed as I think Gunnar fixed it prior to code merge.

# Stretch Goal
Secondary idea I had would be to prototype an alternative solution to the PreviewSql command.  I thought to myself the other day, what if we just make it really easy to inject a MockDbConnection, MockDbDataReader, MockDbCommand, MockDbTransaction?  I have a different approach for this, but I think it would be a big overhaul vs. just stubbing mocks into the framework when someone specifies --preview in the runner.  Plus, in theory, you could add custom interceptors to the mocks to handle various scenarios "as if" you had a real connection, whereas the current no connection logic completely takes that away from you.

When I think of FluentMigrator's weaknesses, this is the big one right now.

# Smaller Bugs

In terms of smaller bugs I personally would love to see fixed:  For SQL Server, the introduction of the Convention Sets for naming conventions didn't fix a scenario where SQL Server auto-generated a primary key name, so the default logic for `Delete.PrimaryKey` assumes it can follow the convention set logic.  Same thing for default constraints and foreign keys. #464 is one ticket but I am confident I created another ticket some place in the last 12 months.

Another relatively easy clean-up ticket: #1310 (Nicer extensions for setting `varchar(max)` for sql server and other dbs, as needing to remember int.MaxValue is kind of weird/magic).

Another small ticket: #1254 fix WriteError behavior (needs tests)

I also haven't merged one PR - Brandon's for FluentMigrator Roslyn Analyzers.  I simply havent had the time to test it locally.  If someone could do that that would be great.

# Home Run
A big, but relatively standalone project, would be: #1222 (FluentMigrator.Shell - run FluentMigrator in Norton Commander mode).

My rationale for wanting this is I personally have a very sophisticated migration process that uses pretty much every FluentMigrator feature.

# Documentation
The Documentation project needs love.  It's good, but it's not where I want it to be, and some stuff, like "migration constraints", isn't explained anywhere.  It would be good to add a page that describes tags, profiles, maintenance migrations, migrations constraints and traits.  We have a ton of different knobs for FluentMigrator now and while I would rather have fewer, it's probably best to document these for consumers so people actually take advantage of them
